[0.128s] Invoking command in '/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing': AMENT_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces:${AMENT_PREFIX_PATH} CMAKE_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces LD_LIBRARY_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/lib:${LD_LIBRARY_PATH} PYTEST_ADDOPTS=--tb=short --junit-xml=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml --junit-prefix=my_testing -o cache_dir=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache -s PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/local/lib/python3.10/dist-packages:${PYTHONPATH} /usr/bin/python3 -m pytest
[0.263s] [1m============================= test session starts ==============================[0m
[0.264s] platform linux -- Python 3.10.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
[0.266s] cachedir: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache
[0.266s] rootdir: /home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing
[0.266s] plugins: launch-testing-1.0.10, ament-lint-0.12.14, ament-copyright-0.12.14, ament-xmllint-0.12.14, launch-testing-ros-0.19.10, ament-pep257-0.12.14, ament-flake8-0.12.14, colcon-core-0.19.0, cov-3.0.0
[0.331s] [1mcollecting ... [0m[1m
[0.331s] collected 1 item                                                               [0m
[0.331s] 
[0.339s] test/test_integration.py [INFO] [launch]: All log files can be found below /home/phr/.ros/log/2025-08-05-22-54-43-382334-PHR-57091
[0.339s] [INFO] [launch]: Default logging verbosity is set to INFO
[0.351s] [INFO] [number_publisher-1]: process started with pid [57105]
[0.351s] [INFO] [number_processor-2]: process started with pid [57107]
[0.376s] test_communication_and_logic (test_integration.TestIntegration) ... [0m[INFO] [1754405683.426759387] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[0.449s] [number_processor-2] [INFO] [1754405683.500070244] [number_processor]: 数字处理节点已启动，等待接收数字...
[0.449s] [number_publisher-1] [INFO] [1754405683.500070244] [number_publisher]: 数字发布节点已启动，将从1开始发布...
[1.444s] [number_publisher-1] [INFO] [1754405684.495113232] [number_publisher]: 发布: 1
[1.445s] [number_processor-2] [INFO] [1754405684.495572716] [number_processor]: 收到: 1
[1.445s] [number_processor-2] [INFO] [1754405684.496163990] [number_processor]: 发布处理后结果: 2
[2.445s] [number_publisher-1] [INFO] [1754405685.495349494] [number_publisher]: 发布: 2
[2.445s] [number_processor-2] [INFO] [1754405685.495556084] [number_processor]: 收到: 2
[2.446s] [number_processor-2] [INFO] [1754405685.496121237] [number_processor]: 发布处理后结果: 4
[3.444s] [number_publisher-1] [INFO] [1754405686.495129870] [number_publisher]: 发布: 3
[3.445s] [number_processor-2] [INFO] [1754405686.495395825] [number_processor]: 收到: 3
[3.445s] [number_processor-2] [INFO] [1754405686.496093956] [number_processor]: 发布处理后结果: 6
[3.446s] [0m[INFO] [1754405686.497098409] [test_subscriber_node]: 检查接收到的消息。 Got: [2, 4, 6][0m
[3.450s] [0m[INFO] [1754405686.501478350] [test_subscriber_node]: 测试结束，销毁测试节点[0m
[3.450s] ok
[3.459s] test_processor_starts_ok (test_integration.TestIntegration) ... [0m[INFO] [1754405686.510479251] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[3.460s] [0m[INFO] [1754405686.511025408] [test_subscriber_node]: 正在检查 processor 节点的启动日志...[0m
[3.461s] [0m[INFO] [1754405686.511729475] [test_subscriber_node]: Processor 节点启动正常。[0m
[3.465s] [0m[INFO] [1754405686.516117726] [test_subscriber_node]: 测试结束，销毁测试节点[0m
[3.465s] ok
[3.473s] test_publisher_starts_ok (test_integration.TestIntegration) ... [0m[INFO] [1754405686.523816163] [test_subscriber_node]: 测试节点test_subscriber_node已创建[0m
[3.473s] [0m[INFO] [1754405686.524297018] [test_subscriber_node]: 正在检查 publisher 节点的启动日志...[0m
[3.474s] [0m[INFO] [1754405686.524783810] [test_subscriber_node]: Publisher 节点启动正常。[0m
[3.477s] [0m[INFO] [1754405686.528127240] [test_subscriber_node]: 测试结束，销毁测试节点[0m
[3.477s] ok
[3.477s] 
[3.477s] ----------------------------------------------------------------------
[3.477s] Ran 3 tests in 3.138s
[3.478s] 
[3.478s] OK
[3.479s] [INFO] [number_processor-2]: sending signal 'SIGINT' to process[number_processor-2]
[3.480s] [INFO] [number_publisher-1]: sending signal 'SIGINT' to process[number_publisher-1]
[3.480s] [number_processor-2] Traceback (most recent call last):
[3.480s] [number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor", line 33, in <module>
[3.481s] [number_processor-2]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_processor')())
[3.481s] [number_processor-2]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_processor_node.py", line 32, in main
[3.481s] [number_processor-2]     rclpy.spin(node)
[3.481s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[3.482s] [number_processor-2]     executor.spin_once()
[3.482s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[3.482s] [number_processor-2]     self._spin_once_impl(timeout_sec)
[3.482s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[3.483s] [number_processor-2]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[3.483s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[3.483s] [number_publisher-1] Traceback (most recent call last):
[3.483s] [number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher", line 33, in <module>
[3.483s] [number_publisher-1]     sys.exit(load_entry_point('my-algorithms==0.0.0', 'console_scripts', 'number_publisher')())
[3.483s] [number_publisher-1]   File "/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages/my_algorithms/number_publisher_node.py", line 25, in main
[3.483s] [number_publisher-1]     rclpy.spin(node)
[3.483s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/__init__.py", line 229, in spin
[3.483s] [number_publisher-1]     executor.spin_once()
[3.483s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 751, in spin_once
[3.484s] [number_processor-2]     return next(self._cb_iter)
[3.484s] [number_processor-2]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[3.484s] [number_processor-2]     wait_set.wait(timeout_nsec)
[3.484s] [number_processor-2] KeyboardInterrupt
[3.484s] [number_publisher-1]     self._spin_once_impl(timeout_sec)
[3.484s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 740, in _spin_once_impl
[3.484s] [number_publisher-1]     handler, entity, node = self.wait_for_ready_callbacks(timeout_sec=timeout_sec)
[3.484s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 723, in wait_for_ready_callbacks
[3.485s] [number_publisher-1]     return next(self._cb_iter)
[3.485s] [number_publisher-1]   File "/opt/ros/humble/local/lib/python3.10/dist-packages/rclpy/executors.py", line 620, in _wait_for_ready_callbacks
[3.485s] [number_publisher-1]     wait_set.wait(timeout_nsec)
[3.485s] [number_publisher-1] KeyboardInterrupt
[3.508s] [ERROR] [number_processor-2]: process has died [pid 57107, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_processor --ros-args -r __node:=number_processor'].
[3.510s] [ERROR] [number_publisher-1]: process has died [pid 57105, exit code -2, cmd '/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/my_algorithms/number_publisher --ros-args -r __node:=number_publisher'].
[3.510s] 
[3.510s] ----------------------------------------------------------------------
[3.510s] Ran 0 tests in 0.000s
[3.511s] 
[3.511s] OK
[3.519s] [32m.[0m
[3.519s] 
[3.519s] - generated xml file: /home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml -
[3.519s] [32m============================== [32m[1m1 passed[0m[32m in 3.26s[0m[32m ===============================[0m
[3.682s] Invoked command in '/home/phr/vscode_project/ros2_launch_testing_ws/src/my_testing' returned '0': AMENT_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces:${AMENT_PREFIX_PATH} CMAKE_PREFIX_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces LD_LIBRARY_PATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/lib:${LD_LIBRARY_PATH} PYTEST_ADDOPTS=--tb=short --junit-xml=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/pytest.xml --junit-prefix=my_testing -o cache_dir=/home/phr/vscode_project/ros2_launch_testing_ws/build/my_testing/.pytest_cache -s PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=/home/phr/vscode_project/ros2_launch_testing_ws/install/my_testing/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_algorithms/lib/python3.10/site-packages:/home/phr/vscode_project/ros2_launch_testing_ws/install/my_interfaces/local/lib/python3.10/dist-packages:${PYTHONPATH} /usr/bin/python3 -m pytest
